tests/test_backward_compatibility.py

```
@@ -0,0 +1,534 @@
+            
+            # Verify new MCP tools are present
+            new_tools = [
+                "get_toolpath_passes",
+                "get_toolpath_linking",
+                "analyze_toolpath_sequence"
+            ]
+            
+            for tool in new_tools:
+                assert tool in tool_names, f"New MCP tool {tool} not found in registered tools"
+                
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)